doc/rfcs/001-allow-guests-to-open-multiple-remote-buffers.md

```diff
@@ -0,0 +1,55 @@
+
+## Status
+
+This is a proposal and is not yet implemented.
+
+## Summary
+
+Allow a separate guest editor for each open buffer in the host's workspace.
+
+## Motivation
+
+Currently, guests can only view host buffers through a specialized "portal editor" that automatically switches its contents to track the active editor of the host. This automatic switching can interrupt guests in the middle of typing. It also makes it impossible for guests to edit any buffer other than the one the host is actively editing.
+
+Ideally, guests would be able to edit any file in the host's current project, but that will require replication of the file-system which is beyond the scope of this proposal. We can get much closer to that ideal by giving guests access to any open buffer in the host's workspace. This can be implemented without substantial changes to the current replication architecture.
+
+## Explanation
+
+### Remote buffers
+
+Just as guests can open editors for local buffers corresponding to files on their local file system, they can also open editors for *remote buffers* corresponding to buffers in any remote workspace that they have joined.
+
+### Following
+
+When you first join a portal as a guest, you're automatically *following* the host. This means that you jump to the host's current position and continue to follow them as they move between buffers.
+
+In order to jump to the host's current position, an editor is automatically added to your workspace for their active buffer when you first join their workspace. If you continue to follow the host, any time they switch to a new buffer, a new editor for that remote buffer is automatically added to your workspace and focused. Existing editors for previous remote buffers are not automatically closed when this switch occurs.
+
+You can follow any other guest participating in the host's workspace in the exact same way. If they move between buffers, you will follow them. The host does not enjoy any special privilege with respect to the ability to be followed between different files.
+
+Editors for remote buffers are *only* automatically opened when you are following another collaborator. If you are not following someone, no editors are automatically opened. When you start following another collaborator again, an editor will be automatically opened based on their location. You can also open any buffer in the host's workspace directly by navigating to it...
+
+### Navigation
+
+As a guest of a shared workspace, you don't automatically receive access to the host's file system. Tools like the tree view, the file finder, and project search don't have access to their current project. (This capability will be the subject of an upcoming RFC).
+
+For now, however, the file finder *will* be augmented to make it easy for you to navigate any buffer that the host currently has open in their workspace. These remote buffers will be decorated with the avatar of the host that owns their workspace.
```

Comment:
> These remote buffers will be decorated with the avatar of the host that owns their workspace.

Nice. Do you have any particular type of decoration in mind?

Reply:
Will custom icon packages be prevented from overriding this avatar? I can see it being problematic if this isn't prevented and these remote tabs aren't yet separated.

Reply:
Thanks for the feedback, @Arcanemagus. 

We're opting to let the RFC remain vague regarding the specific UI design details so that we can explore various options. However, the changes should be additive in nature (i.e., not removing/replacing any existing UI components).